book/src/storage/transfer-to-object.md

```diff
@@ -1,4 +1,132 @@
+
+An object can be transferred to another object by calling the `transfer` function, or its public
+version - `public_transfer`. The behaviour is identical to the transfer to account.
+
+```move
+// File: sui-framework/sources/transfer.move
+module sui::transfer;
+
+public fun transfer<T: key>(object: T, recipient: address) {}
+public fun public_transfer<T: key + store>(object: T, recipient: address) {}
+```
+
+## Receiving Objects
+
+While objects _owned by accounts_ can be used in the transaction directly (given that the sender of
+the transaction is the owner), objects transferred to other objects first need to be _received_
+through their owner. The owner object needs to implement a custom function that will call the
+`receive` function from the `transfer` module.
+
+```move
+// File: sui-framework/sources/transfer.move
+module sui::transfer;
+
+/// Special type which cannot be constructed in Move but can be passed as a
+/// special input to a PTB.
+public struct Receiving<phantom T: key> has drop {
+    id: ID,
+    version: u64,
+}
+
+/// Private receiving function for `key`-only objects.
+public fun receive<T: key>(parent: &mut UID, to_receive: Receiving<T>): T { /* ... */ }
+
+/// Public receiving function for `key + store` objects.
+public fun public_receive<T: key + store>(parent: &mut UID, to_receive: Receiving<T>): {
+  /* ... */
+}
+```
+
+The `Receiving` type is a special type that references an object-owned object. It cannot be
+constructed in Move, and can only be passed as a special input to a
+[Programmable Transaction Block (PTB)](../concepts/what-is-a-transaction.md). The `receive` (or
+`public_receive`) function must be called in order to exchange the `Receiving` instance for the
+actual object.
+
+While this may seem overwhelming, we will demonstrate the feature in a simple example.
+
+## Example
+
+To demonstrate the feature, let's consider a simple `PostOffice` object that can receive any objects
+transferred to it. Given that the transfer itself can be performed on the PTB level, or in a Move
+function, we will focus on the receiving part.
+
+```move
+module book::post_office;
+
+// While the `sui::transfer` module is implicitly imported, the `Receiving` type
+// is not. We need to import it explicitly.
+use sui::transfer::Receiving;
+
+/// The `PostOffice` object that can receive any objects transferred to it via
+/// a custom `receive` function.
+public struct PostOffice has key {
+    id: UID,
+}
+
+/// Voila! The `receive` function can now be called on the `PostOffice` object
+/// to receive any objects transferred to it.
+public fun public_receive<T: key + store>(
+    po: &mut PostOffice,
+    to_receive: Receiving<T>
+): T {
+    transfer::public_receive(&mut po.id, to_receive)
+}
+
+/// For objects without `store` (though, they would require special handling), we
+/// can implement the non-public version of the `receive` function.
+public fun receive<T: key>(po: &mut PostOffice, to_receive: Receiving<T>): T {
```

Comment:
I'm very confused about the non-public receive here. I don't understand the connection between whether or not something has `store` and whether the `receive` function is public. I also don't have a clear picture of which I should use under what conditions.